src/script.js

```diff
@@ -0,0 +1,53 @@
+function waitFor(element, eventName) {
+  return new Promise((resolve) => {
+    // eslint-disable-next-line no-shadow
+    const handler = (event) => {
+      if (event.target === element) {
+        element.removeEventListener(eventName, handler);
+
+        resolve(
+          `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`,
+        );
+      }
+    };
+
+    // Check if the event has already happened
+    if (
+      eventName === 'DOMContentLoaded' &&
+      document.readyState === 'complete'
+    ) {
+      resolve(
+        `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`,
+      );
+    } else {
+      element.addEventListener(eventName, handler);
+    }
```

Comment:
The condition to check if the `eventName` is `DOMContentLoaded` and if the `document.readyState` is `complete` is not needed here. The `waitFor` function is designed to handle any event on any element, not just the `DOMContentLoaded` event. This specific logic should probably be handled outside of this function.